spec/registry.md

```diff
@@ -559,6 +559,175 @@ together with the resolved options' values.
 
 The _function_ `:currency` performs selection as described in [Number Selection](#number-selection) below.
 
+### The `:unit` function
+
+The _function_ `:unit` is a selector and formatter for unitized values,
+that is, numeric values associated with a unit of measurement.
+This is a specialized form of numeric selection and formatting.
+
+> [!IMPORTANT]
+> Implementation of this function is **_OPTIONAL_**.
+> Any implementation of this function is strongly encouraged to follow this specification.
+
+#### Operands
+
+The _operand_ of the `:unit` function can be one of any number of
+implementation-defined types,
+each of which contains a numerical `value` plus a `unit`
+or it can be a [Number Operand](#number-operands), as long as the _option_
+`unit` is provided.
+The _option_ `unit` MAY be used to override the units of an implementation-defined type,
+provided the units are compatible and the implementation supports conversion.
+
+The value of the _operand_'s `unit` SHOULD be either a string containing a
+valid [Unit Identifier](https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers)
+or an implementation-defined unit type.
+
+A [Number Operand](#number-operands) without a `unit` _option_ results in a _Bad Operand_ error.
+
+> [!NOTE]
+> For example, in ICU4J, the type `com.ibm.icu.util.Measure` might be used
+> as an _operand_ for `:unit` because it contains the `value` and `unit`.
+
+> [!NOTE]
+> For runtime environments that do not provide a ready-made data structure,
+> class, or type for unit values, the implementation ought to provide
+> a data structure, convenience function, or documentation on how to encode
+> the value and unit for formatting.
+> For example, such an implementation might define a "unit operand"
+> to include a key-value structure with specific keys to be the
+> local unit operand, which might look like the following:
+> ```
+> {
+>    "value": 123.45,
+>    "unit": "kilometer-per-hour"
+> }
+> ```
+
+#### Options
+
+Some _options_ do not have default values defined in this specification.
+The defaults for these _options_ are implementation-dependent.
+In general, the default values for such _options_ depend on the locale, 
+the unit,
+the value of other _options_, or all of these.
+
+> [!NOTE]
+> The option `select` does not accept the value `ordinal` because selecting
+> unit values using ordinal rules makes no sense.
+
+The following options and their values are required to be available on the function `:unit`:
+- `select`
+   -  `plural` (default)
+   -  `exact`
+- `unit`
+   - valid [Unit Identifier](https://www.unicode.org/reports/tr35/tr35-general.html#unit-identifiers)
+     (no default)
```

Comment:
I continue to be uncomfortable with an effective requirement to emit Unsupported Operation errors for this one option instead of Bad Option, for values that are not supported by the implementation. That feels like a wart.

Reply:
The definitions of [your engagement is non-provisioned](https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md#bad-option) and https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md#bad-option overlap. Thus if something is in the intersection one could go either way. This needs to be cleaned up in the future, but I added a suggestion to change to Bad Option to resolve this.

Reply:
I addressed this by making _Unsupported Operation_ SHOULD for conversions that the implementation doesn't not handle, which might include irrational ones ("milliliters => furlongs"). I removed all other mention of Unsupported or Bad Option to allow the default stuff for functions to cover it.